src/irmin/irmin.mli

```diff
@@ -434,19 +434,23 @@ module Type: sig
   type 'a size_of = 'a -> [`Size of int | `Buffer of string]
     (** The type for size function related to binary encoder/decoders. *)
 
-  val encode_bin: ?buf:(bytes * int) -> 'a t -> 'a -> string
-  (** [encode_bin t e] encodes [t] into a [string] buffer. The size
-     of the returned buffer is precomputed and the buffer is allocated
-     at once or it can be passed using the optional argument [buf].
+  val encode_bin: 'a t -> 'a encode_bin
+  (** [encode_bin t] is the binary encoder for values of type [t]. *)
 
-      {b NOTE:} There is a special case when the parameter [t] is a
-     single buffer (of type [bytes] or [string]): the original value
-     is returned as is, without being copied. *)
+  val decode_bin: 'a t -> 'a decode_bin
+  (** [decode_bin t] is the binary decoder for values of type [t]. *)
 
-  val decode_bin: ?exact:bool -> ?off:int -> 'a t ->
+  val to_bin_string: 'a t -> 'a -> string
+  (** [to_bin_string t x] use {!encode_bin} to convert [x] to a string.
+
+      {b NOTE:} When the parameter [t] is a single buffer (of type
+      [bytes] or [string]) the original buffer is returned without
+      being copied. *)
```

Comment:
I don't follow the documentation sttring here, what is `t` and what is `x`? and is something allocated or not?